libs/s25main/Game.cpp

```diff
@@ -32,15 +30,7 @@ void Game::Start(bool startFromSave)
     if(startFromSave)
         CheckObjective();
     else
-    {
-        if(ggs_.objective == GameObjective::EconomyMode)
```

Comment:
we probably do not want that the addon is removed?

Reply:
It was moved to GameClient, but it should stay here

Reply:
First of all thank you for looking at it thoroughly and for all your suggestions.

I moved the Econhandler initialization to the GameClient because Game::Start() is called too late for the dskGameInterface to construct the economic progress window when the game interface is constructed. 
I also think it makes more sense to initialize this with the gameclient already as that is also where the gameworld setup is being done and where we read savegame snapshots which can also lead to creation of the econhandler.

Reply:
I now moved this back here and construct the windows after game start as suggested by @Flamefire [here](https://github.com/Return-To-The-Roots/s25client/pull/1432#pullrequestreview-737034812).